### PR TITLE
Add support for Debian 9, drop debian 8

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,14 +33,19 @@ matrix:
...
-  - rvm: 2.4.0
+  - rvm: 2.4.2
     bundler_args: --without development
-    env: PUPPET_VERSION="~> 4.0" CHECK=acceptance BEAKER_set="docker/centos-7" bundle exec rake acceptance
+    env: PUPPET_VERSION="~> 5.0" CHECK=acceptance BEAKER_set="docker/centos-7" bundle exec rake acceptance
+    services: docker
+    sudo: required
+  - rvm: 2.4.2
+    bundler_args: --without development
+    env: PUPPET_VERSION="~> 5.0" CHECK=acceptance BEAKER_set="docker/debian-9" bundle exec rake acceptance
```
Comment (ekohl) [inline .travis.yml]: ```beaker-puppet_install_helper``` uses a different syntax. How this even worked in the past is a mystery to me but if you read https://github.com/puppetlabs/beaker-puppet_install_helper#run_puppet_install_helper then it suggests you need explicit version numbers.

**Comment by ekohl** [comment]
I honestly think setting Puppet versions is pretty broken with the acceptance tests. I'd much rather set the repo collection to pc1 or puppet5. For not just leave it out and use the default so we at least test something on debian 5?

**Comment by bastelfreak** [comment]
Current state: In general everything is working. Also this PR fixes https://github.com/voxpupuli/puppet-gluster/issues/140. Debian 8 seems to be a bit strange. After the package installation the daemon is started, but not via unit file (which is also missing in 3.8, but that is EOL anyway)? So puppet tries to start the daemon, bind to the tcp socket fails because it is already running -> everything is broken. Personally I would drop Debian 8 support. It is legacy, Debian 9 is working. Objections?

**Comment by bastelfreak** [comment]
Also the unit files are really really bad. They still rely on the legacy sysv scripts. The mentionted issues result in the usage of official glusterfs repos on Debian 8. Maybe it works if we use the debian-own packages for glusterfs, on debian 8 only. But implementing that would be really really painful and adds a lot of additional logic to the modul. Also the shipped glusterfs version in Debian 8 is, of course, EOL.

**Comment by ekohl** [comment]
👍 to dropping Debian Jessie.

File: .travis.yml (hunk 1 above)
Comment (bastelfreak) [reply .travis.yml]: to clarify: This never worked and was a copy and paste issue

**Comment by ekohl** [comment]
This need a rebase now. Could you also add the nodesets to ```.sync.yml```?

**Comment by bastelfreak** [comment]
I also dropped debian 8 from the metadata.json. We already dropped official support for it and purged it from the travis acceptance test matrix. We just forgot the entry in the metadata.json

**Pull request merged**